src/razor/src/blazorDebug/blazorDebugConfigurationProvider.ts

```diff
@@ -179,6 +263,228 @@ export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurati
+                    );
+                    const oldPid = BlazorDebugConfigurationProvider.pidsByUrl.get(urlStr);
+                    if (oldPid != undefined) {
+                        process.kill(oldPid);
+                    }
+                    BlazorDebugConfigurationProvider.pidsByUrl.set(urlStr, spawnedProxy.pid);
+                    const url = new URL(proxyUrlString);
+                    newUri = `${url.protocol.replace(`http`, `ws`)}//${url.host}{browserInspectUriPath}`;
+                    eventEmmiter.emit('vsWebAssemblyReady');
+                }
+            });
+
+            stream.on('err', (err) => {
+                BlazorDebugConfigurationProvider.vsWebAssemblyBridgeOutputChannel.appendLine(err.toString());
+                eventEmmiter.emit('vsWebAssemblyReady');
+            });
+        }
+
+        handleData(spawnedProxy.stdout);
+        handleData(spawnedProxy.stderr);
+        await new Promise((resolve) => eventEmmiter.once('vsWebAssemblyReady', resolve));
+        return [newUri, proxyICorDebugPort, proxyBrowserPort];
+    }
+
+    private static async isNet9OrNewer(projectPath: string): Promise<boolean> {
+        let isNet9OrNewer = false;
+        const configurationsObject = {
+            $properties: ['BuildProperties'],
+            BuildProperties: {
+                $properties: ['Id', 'Name', 'Value', 'StorageType'],
+                $where: {
+                    '&&': [
+                        { '==': [{ '.': 'Name' }, 'TargetFramework'] },
+                        { '==': [{ '.': 'StorageType' }, 'ProjectFile'] },
+                    ],
+                },
+            },
+        };
+        const query = {
+            context: 'Projects',
+            query: {
+                $properties: ['ActiveConfigurations', 'Path'],
+                ActiveConfigurations: configurationsObject,
+                $filter: { ProjectsByCapabilities: ['WebAssembly'] },
+            },
+        };
+        let queryString = '';
+        if (projectPath != '') {
+            const query = {
+                context: 'Projects',
+                query: {
+                    $properties: ['ActiveConfigurations', 'Path'],
+                    ActiveConfigurations: configurationsObject,
+                    $where: { '==': [{ '.': 'Path' }, projectPath] },
+                },
+            };
+            queryString = JSON.stringify(query);
+        } else {
+            queryString = JSON.stringify(query);
+        }
+        const proxy = await getCSharpDevKit()?.exports.serviceBroker.getProxy<IQueryExecutionService>(
+            Descriptors.projectQueryExecutionService
+        );
+        if (!proxy) {
+            throw new Error('Unable to obtain required service from C# Dev Kit.');
+        }
+        try {
+            const result = await proxy.ExecuteQueryAsync(queryString);
+            const queryResult = JSON.parse(result) as QueryResult<Project>;
+            const pattern = /net(\d+\.\d+)/;
```

Comment:
Should this be `/^net(\d+\.\d+)$/`? Or maybe `\b` instead of `$` if you want to support one of the `-` platform things.

Reply:
done.